addons/website/static/src/snippets/s_popup/000.js

```diff
@@ -39,21 +45,27 @@ const PopupWidget = publicWidget.Widget.extend({
      * @private
      */
     _bindPopup: function () {
-        const $main = this.$target.find('.modal');
-
-        let display = $main.data('display');
-        let delay = $main.data('showAfter');
 
         if (config.device.isMobile) {
-            if (display === 'onExit') {
-                display = 'afterDelay';
-                delay = 5000;
+            if (this.display === 'mouseExit') {
+                this.display = 'afterDelay';
+                this.delay = 5000;
             }
             this.$('.modal').removeClass('s_popup_middle').addClass('s_popup_bottom');
         }
 
-        if (display === 'afterDelay') {
-            this.timeout = setTimeout(() => this._showPopup(), delay);
+        if (this.display === 'afterDelay') {
+            this.timeout = setTimeout(() => this._showPopup(), this.delay);
+        } else if (this.display === 'onClick') {
+            let anchor = $.escapeSelector(this.$target.find('.modal').attr('id'));
+
+            const classesToAdd = {
+                'data-toggle': this.editableMode ? null : 'modal',
+                'data-target': this.editableMode ? null : '#' + anchor,
+            };
+            const $linkSelector = "a[href*='" + anchor + "']";
+            const $link = $($linkSelector);
+            $link.attr(classesToAdd);
```

Comment:
I am not sure to understand this... should not this be the role of the option and not the public widget ? I think this widget should not be enabled in edit mode (like before).

Reply:
@IT-Ideas Hi ;) Could you explain why you made it like that please ?

Reply:
Hey! @bvr-odoo , I guess because I did not know about the snippet architecture and design at that time...

Reply:
@qsm-odoo Are you agree that it should be done with the link dialog (on save) ? Checking if the link is a modal and if yes adding the attributes

Reply:
Yes indeed ;)